chain/beacon/node.go

```diff
@@ -307,7 +314,14 @@ func (h *Handler) Reset() {
 // run will wait until it is supposed to start
 func (h *Handler) run(startTime int64) {
 	chanTick := h.ticker.ChannelAt(startTime)
-	h.l.Debugw("", "run_round", "wait", "until", startTime)
+
+	// we cancel the context for any existing handler runs and replace it with a new one
+	select {
+	case <-h.conf.Clock.After(time.Unix(startTime, 0).Sub(h.conf.Clock.Now())):
+		h.killRunInFlight <- true
+	default:
+		h.l.Debugw("", "run_round", "wait", "until", startTime, "now", h.conf.Clock.Now().Unix())
+	}
```

Comment:
the `h.killRunInFlight` is never called because the `default` statement is always taken before the `After` timer fires

Reply:
Solved in #1260